_posts/2020-08-01-Making-IrishGen-SPARQL-Part-III.md

```diff
@@ -0,0 +1,144 @@
+the user.  What is returned by a Triplestore is undefined by SPARQL
+(and deliberately so) in the standard.  What is generally returned is
+the RDF that the Triplestore has about a particular URL.  This is
+useful since, if the user knows the URL, they can interrogate the
+Triplestore about what is stored in the Triplestore about that URL
+without needing to go through the ceremony of a construct query.
+Moreover, Describe queries are often used in visual representations of
+RDF graphs.
+
+For instance, the screenshot below is from GraphDB's "Visual Graph"
+feature and uses Finn mac Cumaill as the exemplar.  What happens is
+that the user sends the URL in which they are interested, GraphDB then
+makes a Describe query on the URL, and RDF returned is used by GraphDB
+to create a visual representation.  This is very useful when
+investigating single individuals within IrishGen.  Additionally, it is
+good if a user wants to have a secure, single starting point to
+explore the graph in various ways.
+
+<img src="{{site.baseurl}}/assets/images/describe_part3.png" />
+
+Using the "Expand" feature of GraphDB's Visual Graph will send further
+Describe queries to the Triplestore.  The resulting RDF will then be
+rendered in visual form and any connections will be presented to the
+user.  Describe queries can also be done as plain SPARQL as well.  For
+instance:
+
+```sparql
+describe <http://example.com/LL/lagin.trig#Find>
+```
+
+will return over 1000 results giving everything that the Triplestore
+contains about Find.  The result of this is that the Visual Graph can
+be overwhelmed with information.  The best practice here is to choose
+the predicates that the user is interested in rather than attempting
+to sift through all the information.  While manipulating the Visual
+Graph feature of GraphDB is outside the scope of the present post,
+most often a user is only interested in one or two predicates so using
+the Visual Graph settings to filter out all results other than the
+ones that the user are interested in.
+
+# Ask
+
+The Ask query has one use to inform the user if a basic graph pattern
+has a solution or not.  No information other than a boolean that
+indicates existence or otherwise of a solution is returned.
+
+In terms of IrishGen, this query form is used to ask for a quick
+indication if the information that a user is looking for exists within
+IrishGen.  There is little point in using this in a SPARQL query when
+using IrishGen.  A user will be better served by using a Select or
+Construct query as usually a user will need the results of a query
+rather than mere existence of a solution.  However, as an example:
+
+```sparql
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+ask
+from named <http://example.com/LL>
+where {
+    graph ?g {
+    	?a irishRel:nomName "Find"
+    }
+}
+```
+
+will return `YES` as there is, in fact, a solution to the given
+basic graph pattern.
```

Comment:
Perhaps explain what you were asking and what the YES means, for the completely uninitiated?